Shopipy.Web/Controllers/BaseController.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.AspNetCore.Mvc;
+using Shopipy.Web.Services;
+using Shopipy.UserManagement.Dtos;
+using Shopipy.Web.ViewModels;
+
+namespace Shopipy.Web.Controllers;
+
+public abstract class BaseController(UserService userService) : Controller
+{
+
+    private UserResponseDto? _currentUser;
+
+    protected async Task<UserResponseDto?> GetCurrentUserAsync()
+    {
+        if (_currentUser != null) return _currentUser;
+
+        var userId = HttpContext.Items["UserId"]?.ToString();
+        var token = HttpContext.Request.Cookies["BearerToken"];
+
+        if (!string.IsNullOrEmpty(userId) && !string.IsNullOrEmpty(token))
+        {
+            _currentUser = await userService.GetUserByIdAsync(userId, token);
+        }
+
+        return _currentUser;
+    }
+
+    public override ViewResult View(string? viewName, object? model)
+    {
+        if (model is BaseViewModel viewModel)
```

Comment:
Could you replace this with [result filter usage](https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-9.0#result-filters)? Filter would check if model is instance of `BaseViewModel` and would call `GetUserByIdAsync`.

Reply:
Aside from that I think, that not that many views actually need the user data, user data may be needed before, during service stage, so I think both `BaseController` and `BaseModel` are likely not necessary.

Reply:
every view that uses regular layout needs the user for the authentication part in navbar atm

Reply:
Then make navbar a view component

Reply:
noidontthinkso